File: Codes/3375-minimum-operations-to-make-array-values-equal-to-k.py
'''
Author: LetMeFly
Date: 2025-04-09 22:09:48
LastEditors: LetMeFly.xyz
LastEditTime: 2025-04-09 22:10:06
'''
from typing import List

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        nums.sort(reverse=True)
        if nums[-1] < k:
            return -1
        ans = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                ans += 1
        return ans + (nums[-1] != k)

File: Codes/3375-minimum-operations-to-make-array-values-equal-to-k.go
/*
 * @Author: LetMeFly
 * @Date: 2025-04-09 22:16:49
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-09 22:18:14
 */
package main

import "slices"

func minOperations(nums []int, k int) (ans int) {
    slices.Sort(nums)
    if nums[0] < k {
        return -1
    }
    for i := len(nums) - 1; i > 0; i-- {
        if nums[i] != nums[i - 1] {
            ans++
        }
    }
    if nums[0] != k {
        ans++
    }
    return
}